plugins/by-name/nvim-snippets/default.nix

```
@@ -41,7 +41,8 @@ lib.nixvim.plugins.mkNeovimPlugin {
     '';
 
     search_paths =
-      defaultNullOpts.mkListOf types.str [ { __raw = "vim.fn.stdpath('config') .. '/snippets'"; } ]
+      defaultNullOpts.mkListOf types.str
+        (lib.nixvim.literalLua "{ vim.fn.stdpath('config') .. '/snippets' }")
```

```suggestion
      defaultNullOpts.mkListOf types.str
        [ (lib.nixvim.literalLua "vim.fn.stdpath('config') .. '/snippets'" ])
```
`mkListOf types.str` options allow `__raw` elements by default.

---

> Actually, if we were to start making the default config checks automatically based on pluginDefaults, they would be broken by this PR. I'll turn this into a draft PR.

There's no need to block this PR on tangential or unrelated refactoring.

`pluginDefault` is purely for documentation. There are no plans to use it for generating tests.

Feel free to undraft.

---

@GaetanLepage 's suggestion of `[ (lib.nixvim.literalLua "vim.fn.stdpath('config') .. '/snippets'" ])` won't work, because it's embedding a literalExpression within a value.

You could do any of:

```nix
# I think this is what Gaétan meant:
[ (nestedLiteralLua "...") ]
```

```nix
# This is the least "magic":
literalExpression ''[ (lib.nixvim.mkRaw "...") ]''
```

```nix
# This has a disadvantage of not demonstrating how list items can be raw values:
literalLua "{ ... }"
```

---

> There's no need to block this PR on tangential or unrelated refactoring.
> 
> `pluginDefault` is purely for documentation. There are no plans to use it for generating tests.
> 
> Feel free to undraft.

Great! I thought that was the reason @GaetanLepage suggested a list at the top level.